app/src/main/java/com/cmput301w23t40/capturetheqr/QRCodeList.java

```diff
@@ -20,12 +21,25 @@ public class QRCodeList extends RecyclerView.Adapter<QRCodeList.RecyclerViewHold
 
     private ArrayList<QRCode> qrCodes;
     private Context context;
+    private static OnItemClickListener onItemClickListener;
 
     public QRCodeList(Context context, ArrayList<QRCode> qrCodes){
         this.qrCodes = qrCodes;
         this.context = context;
     }
 
+    public QRCode getCode(int position) {
+        return qrCodes.get(position);
+    }
+
+    public void setOnItemClickListener(OnItemClickListener onItemClickListener) {
+        this.onItemClickListener =  onItemClickListener;
+    }
+
+    public interface OnItemClickListener {
+        void onItemClick(View view, int position);
+    }
```

Comment:
Maybe move this near the top of the class so that anyone reading the file is made aware of the type before it is used

Reply:
fixed!